modules/farm_fd2/src/entrypoints/transplanting/lib.submit.unit.cy.js

```diff
@@ -0,0 +1,221 @@
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    bedFeet: 100,
+    bedWidth: 60,
+    rowsPerBed: '1',
+    equipment: ['Tractor'],
+    depth: 6,
+    speed: 5,
+    comment: 'A comment',
+  };
+
+  let cropToTerm = null;
+  let result = null;
+  let fieldMap = null;
+  let bedMap = null;
+
+  before(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+
+    cy.wrap(farmosUtil.getFieldIdToAssetMap()).then((map) => {
+      fieldMap = map;
+    });
+
+    cy.wrap(farmosUtil.getBedIdToAssetMap()).then((map) => {
+      bedMap = map;
+    });
+
+    cy.wrap(farmosUtil.getCropNameToTermMap()).then((map) => {
+      cropToTerm = map;
+    });
+
+    cy.wrap(traySeedingLib.submitForm(traySeedingForm), { timeout: 10000 })
+      .then(() => {
+        return cy.wrap(farmosUtil.getSeedlings(traySeedingForm.cropName), {
+          timeout: 10000,
+        });
+      })
+      .then((res) => {
+        form.picked[0] = { trays: 25, data: res[res.length - 1] };
+        return cy.wrap(lib.submitForm(form), { timeout: 10000 });
+      })
+      .then((res) => {
+        result = res;
+      });
+  });
+
+  beforeEach(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+  });
+
+  afterEach(() => {
+    cy.saveLocalStorage();
+    cy.saveSessionStorage();
+  });
+
+  it('Check the parents', () => {
+    expect(result.parents[0].id).to.equal(form.picked[0].data.asset_uuid);
+    expect(result.parents[0].type).to.equal('asset--plant');
+    expect(result.parents[0].attributes.name).to.equal(
+      form.picked[0].data.date + '_' + form.picked[0].data.crop
+    );
+  });
+
+  it('Check the tray inventory quantity--standard assets', () => {
+    expect(result.trayInventoryQuantities[0].attributes.value.decimal).to.equal(
+      form.picked[0].trays
+    );
+    expect(result.trayInventoryQuantities[0].type).to.equal(
+      'quantity--standard'
+    );
+    expect(result.trayInventoryQuantities[0].attributes.measure).to.equal(
+      'count'
+    );
+    expect(result.trayInventoryQuantities[0].attributes.label).to.equal(
+      'Trays'
+    );
+    expect(
+      result.trayInventoryQuantities[0].attributes.inventory_adjustment
+    ).to.equal('decrement');
+  });
+
+  it('Check the asset--plant', () => {
+    // Check the plant asset.
+    expect(result.transplantingPlantAsset.type).to.equal('asset--plant');
+    expect(result.transplantingPlantAsset.attributes.name).to.equal(
+      form.transplantingDate + '_' + form.cropName
+    );
+
+    expect(result.transplantingPlantAsset.attributes.status).to.equal('active');
+    expect(result.transplantingPlantAsset.attributes.notes.value).to.equal(
+      form.comment
+    );
+
+    expect(
+      result.transplantingPlantAsset.relationships.plant_type[0].type
+    ).to.equal(cropToTerm.get(form.cropName).type);
+
+    expect(
+      result.transplantingPlantAsset.relationships.plant_type[0].id
+    ).to.equal(cropToTerm.get(form.cropName).id);
+  });
```

Comment:
We should check that the original plant asset is a parent of this one.  When a transplanting is made, the plant asset from the tray seeding becomes a parent of the plant asset from the transplanting.